src/ServiceInsight/MessageList/MessageListViewModel.cs

```diff
@@ -194,6 +194,11 @@ public async Task Handle(SelectedExplorerItemChanged @event)
             selectedExplorerItem = @event.SelectedExplorerItem;
             ServiceControl = @event.SelectedExplorerItem.GetServiceControlClient(clientRegistry);
 
+            if (Parent.SearchInProgress)
+            {
+                return;
+            }
+
```

Comment:
I do not see `SearchInProgress` set to `true` anywhere. Should it be here?

```suggestion
            Parent.SearchInProgress = true;
```